File: sum-of-left-leaves/sum-of-left-leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: TreeNode) -> int:
        '''Calculates the sum of left tree leaves'''
        the_sum = [0]
        if not root:
            return the_sum
        
        self.dfs(root, False, the_sum)
        return the_sum[0]
        
        
    def dfs(self, root: TreeNode, left: bool, the_sum: list[int]):
        '''DFS down the tree, calculating sum of left children'''
        if not root:
            return
        if not root.left and not root.right and left:
            the_sum[0] += root.val
        self.dfs(root.left, True, the_sum)
        self.dfs(root.right, False, the_sum)

File: sum-of-left-leaves/sum-of-left-leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution 
{
    public int sumOfLeftLeaves(TreeNode root) 
    {
        if (root == null) // No tree exists
            return 0;
        
        return DFS(root, false);
    }
    
    public int DFS(TreeNode root, boolean isLeft)
    {
        if (root == null) //No node exists
            return 0;
        
        if (root.left == null && root.right == null && isLeft)
            return root.val;
        int sum = DFS(root.left, true);
        sum += DFS(root.right, false);
        return sum;
    }
}